### PR TITLE
Added F# Console and Class Library Templates. Closes #785

### LINKED ISSUE #785: Support for F# Projects
Is there a hidden way to create a F# project, if not please add F# support soon!

### PR DESCRIPTION
**Pull request by odytrice**
Fixes #785.

Added the following Project Templates
 - F# Console
 - F# Library Project

Thanks!

/cc
@OmniSharp/generator-aspnet-team-push

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
The console project generator routine is missing `dotnet run` hint:
```
Your project is now created, you can use the following commands to get going
    cd "FSharpConsole"
    dotnet restore
    dotnet build (optional, build will also happen when it's run)
```

**Comment by peterblazejewicz** [comment]
We have to use non-standard `defaultNamespace` key in `project.json`:
https://github.com/OmniSharp/generator-aspnet/blob/master/templates/projects/consoleapp/project.json#L20-L22
So you will have to modify project.json templates to add that keys ast he F# supports namespaces

File: app/index.js [hunk 1]
```diff
@@ -309,8 +321,25 @@ var AspnetGenerator = yeoman.generators.Base.extend({
...
+        this.sourceRoot(path.join(__dirname, '../templates/projects/' + this.type));
+        this.fs.copy(path.join(__dirname, '../templates/gitignore.txt'), this.applicationName + '/.gitignore');
+        this.fs.copyTpl(this.templatePath('Program.fs'), this.applicationName + '/Program.fs', this.templatedata);
+        this.fs.copyTpl(this.templatePath('project.json'), this.applicationName + '/project.json', this.templatedata);
+
+        break;
+
       default:
-        this.log('Unknown project type');
+        this.log('Unknown project type ' + this.type);
```
Comment (peterblazejewicz) [inline app/index.js]: This is outside of PR scope

**Comment by odytrice** [comment]
Making the corresponding changes and swashing the commits

**Comment by peterblazejewicz** [comment]
@odytrice Let me know if you are to make updates. I could merge PR and make corrections tomorrow.

**Comment by odytrice** [comment]
Oh okay. I'll add the Web templates

**Comment by odytrice** [comment]
I'm done with the Updates. These Templates are enough for typical Web Development

**Comment by peterblazejewicz** [comment]
Tested locally all projects.
🚢  💃

**Comment by peterblazejewicz** [comment]
@odytrice 
Can you link the source code you've used to create templates? The `README.md` usually contains a reference to original source or inspiration for template. I'll update project readme.

**Comment by odytrice** [comment]
Yeah it's here https://github.com/odytrice/dotnet-templates 
Do I update the Readme?

**Comment by peterblazejewicz** [comment]
Go ahead!

**Comment by odytrice** [comment]
I've updated the Readme file with the source link

**Pull request merged**